api/email_sender.py

```diff
@@ -51,6 +51,8 @@ def _send_email(self, email_msg):
         if smtp:
             smtp.send_message(email_msg)
             smtp.quit()
+        else:
+            raise Exception("Failed to connect to SMTP server")
```

Comment:
I don't think we need to raise exceptions manually. SMTP internally raises it if connection or login fails.
I'll double check.

Reply:
So I double checked. SMTP automatically raises exceptions when errors are found.

e.g. if email configs are missing:
```
$ ./kci user verify testuser1@gmail.com
Sending verification token to testuser1@gmail.com
Error: 400 Client Error: Bad Request for url: http://172.17.0.1:8001/latest/user/request-verify-token
4 validation errors for EmailSettings
smtp_host
  field required (type=value_error.missing)
smtp_port
  field required (type=value_error.missing)
email_sender
  field required (type=value_error.missing)
email_password
  field required (type=value_error.missing)
```

If SMTP username and password mismatch found:
```
kernelci-api | INFO:     Application startup complete.
kernelci-api | INFO:     172.22.0.1:56144 - "POST /latest/user/request-verify-token HTTP/1.1" 500 Internal Server Error
kernelci-api | ERROR:    Exception in ASGI application
kernelci-api | Traceback (most recent call last):
kernelci-api |   File "/home/kernelci/.local/lib/python3.10/site-packages/fastapi_users/manager.py", line 304, in request_verify
kernelci-api |     await self.on_after_request_verify(user, token, request)
kernelci-api |   File "/home/kernelci/./api/user_manager.py", line 56, in on_after_request_verify
kernelci-api |     self.email_sender.create_and_send_email(subject, content, user.email)
kernelci-api |   File "/home/kernelci/./api/email_sender.py", line 65, in create_and_send_email
kernelci-api |     self._send_email(email_msg)
kernelci-api |   File "/home/kernelci/./api/email_sender.py", line 51, in _send_email
kernelci-api |     smtp = self._smtp_connect()
kernelci-api |   File "/home/kernelci/./api/email_sender.py", line 31, in _smtp_connect
kernelci-api |     smtp.login(self._settings.email_sender,
kernelci-api |   File "/usr/local/lib/python3.10/smtplib.py", line 750, in login
kernelci-api |     raise last_exception
kernelci-api |   File "/usr/local/lib/python3.10/smtplib.py", line 739, in login
kernelci-api |     (code, resp) = self.auth(
kernelci-api |   File "/usr/local/lib/python3.10/smtplib.py", line 662, in auth
kernelci-api |     raise SMTPAuthenticationError(code, resp)
kernelci-api | smtplib.SMTPAuthenticationError: (535, b'5.7.8 Username and Password not accepted. For more information, go to\n5.7.8  https://support.google.com/mail/?p=BadCredentials sg4-20020a17090b520400b0028be1050020sm14445625pjb.29 - gsmtp'
```

If SMTP host is incorrect:
```
kernelci-api | INFO:     Application startup complete.
kernelci-api | INFO:     172.23.0.1:35238 - "POST /latest/user/request-verify-token HTTP/1.1" 500 Internal Server Error
kernelci-api | ERROR:    Exception in ASGI application
kernelci-api |   File "/home/kernelci/.local/lib/python3.10/site-packages/fastapi_users/router/verify.py", line 29, in request_verify_token
kernelci-api |     await user_manager.request_verify(user, request)
kernelci-api |   File "/home/kernelci/.local/lib/python3.10/site-packages/fastapi_users/manager.py", line 304, in request_verify
kernelci-api |     await self.on_after_request_verify(user, token, request)
kernelci-api |   File "/home/kernelci/./api/user_manager.py", line 56, in on_after_request_verify
kernelci-api |     self.email_sender.create_and_send_email(subject, content, user.email)
kernelci-api |   File "/home/kernelci/./api/email_sender.py", line 65, in create_and_send_email
kernelci-api |     self._send_email(email_msg)
kernelci-api |   File "/home/kernelci/./api/email_sender.py", line 51, in _send_email
kernelci-api |     smtp = self._smtp_connect()
kernelci-api |   File "/home/kernelci/./api/email_sender.py", line 25, in _smtp_connect
kernelci-api |     smtp = smtplib.SMTP_SSL(self._settings.smtp_host,
kernelci-api |   File "/usr/local/lib/python3.10/smtplib.py", line 1050, in __init__
kernelci-api |     SMTP.__init__(self, host, port, local_hostname, timeout,
kernelci-api |   File "/usr/local/lib/python3.10/smtplib.py", line 255, in __init__
kernelci-api |     (code, msg) = self.connect(host, port)
kernelci-api |   File "/usr/local/lib/python3.10/smtplib.py", line 341, in connect
kernelci-api |     self.sock = self._get_socket(host, port, self.timeout)
kernelci-api |   File "/usr/local/lib/python3.10/smtplib.py", line 1056, in _get_socket
kernelci-api |     new_socket = super()._get_socket(host, port, timeout)
kernelci-api |   File "/usr/local/lib/python3.10/smtplib.py", line 312, in _get_socket
kernelci-api |     return socket.create_connection((host, port), timeout,
kernelci-api |   File "/usr/local/lib/python3.10/socket.py", line 824, in create_connection
kernelci-api |     for res in getaddrinfo(host, port, 0, SOCK_STREAM):
kernelci-api |   File "/usr/local/lib/python3.10/socket.py", line 955, in getaddrinfo
kernelci-api |     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
kernelci-api | socket.gaierror: [Errno -5] No address associated with hostname
```

Reply:
Please see https://github.com/kernelci/kernelci-api/pull/443. This will enable us to debug failures easily by handling these exceptions properly and making API logs verbose.

Reply:
Thanks, i will drop PR